### PR TITLE
Graph options - add VM and error states 

### PR DESCRIPTION
**Pull request by rudyhuyn**
### Description of the changes:
- Add GraphSettingsViewModel to simplify the UI part of the control and manage all the logic
- Manage error states
- Disable keyboard when the flyout is opened
- Only load GraphSettings after the user clicks on the button to improve launch performance
- Keep default styling of textbox in GraphSettings
- Use TextBlock::Header instead of a TextBlock
- Add a 1px margin between radiobutton/togglebutton to improve the readability of the UI
- Tweak a little the graph buttons

### How changes were validated:
- Manually

### CODE REVIEW

File: src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#include "pch.h"
```
Comment (sanderl) [inline src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.cpp:4]: Needs a copyright header.

File: src/Calculator/GraphingSettingsViewModel.cpp [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#include "pch.h"
```
Comment (sanderl) [inline src/Calculator/GraphingSettingsViewModel.cpp]: do you still need this file?

File: src/Calculator/GraphingSettingsViewModel.h [hunk 3]
```diff
@@ -0,0 +1 @@
+#pragma once
```
Comment (sanderl) [inline src/Calculator/GraphingSettingsViewModel.h]: Do you need this file?

File: src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml.cpp [hunk 4]
```diff
@@ -4,12 +4,15 @@
 #include "CalcViewModel\Common\AppResourceProvider.cpp"
```
Comment (sanderl) [inline src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml.cpp:4]: Copyright header

File: src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml.h [hunk 5]
```diff
@@ -6,7 +6,7 @@
 #pragma once
```
Comment (sanderl) [inline src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml.h:6]: copyright header

**Review by sanderl — CHANGES_REQUESTED** [review]
I did a buddy build and a buddy test. A few things I noticed:
1. For the error state, the background should change to red like the EquationTextBox
2. I noticed sometimes the grid doesn't update according to the options set by the user. 
I graphed sin(x), I noticed the grid options did not reflect what was shown on the graph (it showed -12 and 12 for x and -10 and 10 for y). I pressed enter to get the graph to reset and nothing happened. The graph showed the axes at -1000 to 1000. 

I like that you pulled most of the logic into a VM. I think that is the right design.

I made a few comments about extra files and missing headers.

File: src/Calculator/GraphingSettingsViewModel.cpp (hunk 2 above)
Comment (rudyhuyn) [reply src/Calculator/GraphingSettingsViewModel.cpp]: No

**Comment by rudyhuyn** [comment]
Thank you for your feedback! Copyright added and extra file removed!

I've also fixed some issues with high contrast: ChromeWhiteHighControlSelectedForegroundBrush was missing in HighContrast and the toggle button in GraphControl had some issues.

I've also modified the RadioButton template to use the real ToggleButton brushes, so we will always be consistent with the system (and we won't impact app launch performance with extra brushes).

Should we move this PR of a PR (😂) to the main repository and do a PR instead?

**Pull request closed without merging**